### DISCUSSION TITLE
Psycopg3: Non blocking Async Cursor methods

### DISCUSSION THREAD

**Discussion by khushboovashi** [psycopg/psycopg#346] [Q&A]
Any example of non blocking methods with Async cursor?
How to execute non blocking query with async cursor? As per the documentation async_cursor.execute is a blocking call.

**Comment by DzigaV** [comment]
Look at the examples here: https://www.psycopg.org/psycopg3/docs/advanced/async.html

The condensed syntax is really simple:

```
async with await psycopg.AsyncConnection.connect() as aconn:
    async with aconn.cursor() as cur:
        await cur.execute(...)
```

**Reply by khushboovashi** [reply]
That is blocking Async call, I want non-blocking async call.

**Reply by dvarrazzo** [reply]
Maybe you want a call concurrent with other queries? You should follow an asyncio tutorial.
